### PR TITLE
Anonymized server listing options

### PR DESCRIPTION
**Pull request by Naamloos**
Add two config options controlling the server list: 
one of them disabled querying (drops connection as soon as state 1 (status) is detected
the other disabled the "user list sample" (player counts are still intact, but no players are listed)

### CODE REVIEW

**Comment by Naamloos** [comment]
can someone review pls 🥺

File: Obsidian/Utilities/Config.cs [hunk 1]
```diff
@@ -40,5 +40,9 @@ public class Config : IConfig
         public bool UDPBroadcast = false;
         
         public int PregenerateChunkRange { get; set; } = 15; // by default, pregenerate range from -15 to 15
+
+        public bool AnonymousOnlineList { get; set; } = false;
```
Comment (roxxel) [inline Obsidian/Utilities/Config.cs]: maybe change this to enum?
like
public QueryingType Querying { get; set; }
```cs
enum QueryingType 
{
    Full,
    Anonymized,
    Disabled
}
```

File: Obsidian/Client.cs [hunk 2]
```diff
@@ -149,11 +149,20 @@ public async Task StartConnectionAsync()
...
+                            this.Disconnect();
+                            break;
+                        }
                         switch (id)
                         {
                             case 0x00:
                                 var status = new ServerStatus(Server);
 
+                                if (config.AnonymousOnlineList) // If online list anonymous, clear the player sample.
+                                    status.Players.Sample.Clear();
```
Comment (Craftplacer) [inline Obsidian/Client.cs]: That the player list needs to be cleared seems odd, maybe a change inside ServerStatus might be worth to do.

File: Obsidian/Client.cs [hunk 3]
```diff
@@ -149,11 +149,20 @@ public async Task StartConnectionAsync()
                 switch (this.State)
                 {
                     case ClientState.Status: // Server ping/list
+                        if(!this.config.EnableQuerying)
+                        {
+                            this.Logger.LogInformation("Closing connection, querying is disabled.");
```
Comment (Craftplacer) [inline Obsidian/Client.cs]: Maybe we shouldn't log this, only when in Verbose

**Review by Craftplacer — APPROVED** [review]

File: Obsidian/Client.cs (hunk 3 above)
Comment (Naamloos) [reply Obsidian/Client.cs]: fair, will change

**Comment by Naamloos** [comment]
Resolved all comments.

**Review by Craftplacer — APPROVED** [review]

**Pull request merged**